### PR TITLE
[BugFix] Fix the bug of plan enum mechanism

### LINKED ISSUE #12774: [Bug] Plan enumeration output wrong plan
### Steps to reproduce the behavior (Required)

TPCDS-100g

```sql
select /*+ SET_VAR(cbo_use_nth_exec_plan = 110) */ a.ca_state state, count(*) cnt
 from customer_address a
     ,customer c
     ,store_sales s
     ,date_dim d
     ,item i
 where       a.ca_address_sk = c.c_current_addr_sk
         and c.c_customer_sk = s.ss_customer_sk
         and s.ss_sold_date_sk = d.d_date_sk
         and s.ss_item_sk = i.i_item_sk
         and d.d_month_seq = 
              (select distinct (d_month_seq)
               from date_dim
               where d_year = 2001
                 and d_moy = 1 )
         and i.i_current_price > 1.2 * 
             (select avg(j.i_current_price) 
              from item j 
              where j.i_category = i.i_category)
 group by a.ca_state
 having count(*) >= 10
 order by cnt, a.ca_state 
 limit 100;
```

### Expected behavior (Required)

Right result

### Real behavior (Required)

Wrong result(0 rows)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

```
+-------------------+
| current_version() |
+-------------------+
| UNKNOWN 133fd49 |
+-------------------+
```

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12774

## Problem Summary(Required) ：

Plan enum mechanism use the following data struct to record the crucial information for enumeration
* `Group:: satisfyRequiredPropertyGroupExpressions`, it's type is map, mapping from RequiredProperty to the corresponding best GroupExpression
* `GroupExpression::validOutputInputProperties`, it's type is list, and item type is `OutputInputProperty`, which holds `(RequiredProperty, children's RequiredProperties)` pair

And there are two main errors

1. The purpose of `validOutputInputProperties` is to save `(RequiredProperty, children's RequiredProperties)` pair, but actually, we saved `(RequiredProperty, children's OutputProperties)` instead.
2. Some operators, such as JoinOperator, have extra requirements for its two children, which is written in `ChildOutputPropertyGuarantor`. But when we processing JoinOperator during enumeration, we simply pick one of the ExpressionGroup in the left child and one of the ExpressionGroup in the right child(Cartesian product), and this combination may break the guarantee that `ChildOutputPropertyGuarantor` requires.

## Solution

1. `Group:: satisfyRequiredPropertyGroupExpressions` -> `Group:: satisfyOutputPropertyGroupExpressions`, now it mapping from OutputProperty to the corresponding best GroupExpression
2. `GroupExpression::validOutputInputProperties` -> `GroupExpression::validOutputPropertyGroups `, it's item type is `OutputPropertyGroup`(renaming from `OutputInputProperty`), which holds `(OutputProperty, children's OutputProperties)` pair

The thought is simple, we use OutputProperty instead of RequiredProperty to support plan enumation, and OutputProperty natually respect the requirements of `ChildOutputPropertyGuarantor`, so no extra work is needed

But, the space of enumation may declined.

## Other modifications

Most of the enum plan tests are of wrong format(missing label `[end]` in .`sql` file), and didn't work normally. So I rewrote them all over.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 95 / 99 (95.96%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|17|82.35%|[163, 176, 177]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|8|9|88.89%|[59]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|28|28|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|28|28|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 96 / 100 (96.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|17|82.35%|[163, 176, 177]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|8|9|88.89%|[59]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|27|27|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|28|28|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 96 / 100 (96.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|17|82.35%|[163, 176, 177]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|8|9|88.89%|[59]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|27|27|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|28|28|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 96 / 100 (96.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|17|82.35%|[163, 176, 177]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|8|9|88.89%|[59]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|27|27|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|28|28|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 133 / 139 (95.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|17|82.35%|[163, 176, 177]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|30|30|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|52|52|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 141 / 149 (94.63%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|15|18|83.33%|[162, 175, 176]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|63|65|96.92%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|30|30|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 142 / 149 (95.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|65|67|97.01%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|30|30|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 142 / 149 (95.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|65|67|97.01%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|30|30|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/EnforceAndCostTask.java [hunk 1]
```diff
@@ -444,31 +451,33 @@ private PhysicalPropertySet enforceProperty(PhysicalPropertySet outputProperty,
...
                 context.getRequiredProperty().getDistributionProperty().appendEnforcers(groupExpression.getGroup());
 
         updateCostWithEnforcer(enforcer, oldOutputProperty, newOutputProperty);
+        recordPlanEnumInfo(enforcer, newOutputProperty, isOldEnforced, Lists.newArrayList(oldOutputProperty));
 
         return newOutputProperty;
     }
 
-    private PhysicalPropertySet enforceSort(PhysicalPropertySet oldOutputProperty) {
+    private PhysicalPropertySet enforceSort(boolean isOldEnforced, PhysicalPropertySet oldOutputProperty) {
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/EnforceAndCostTask.java:473]: If only sort or distribution property of output not satisfies the required property, then only one enforce will be performed, `originalOp -> enforceOp`, in this situation, `isOldEnforced = false`, which means `originalOp` is not a enforceOp

if both sort and distribution properties of output not satisfy the required property, then two enforces will be applied, `originalOp -> enforceOp(sort) -> enforceOp(distribution)`
* For the first `enforceOp`, `isOldEnforced=false`
* For the second `enforceOp`, `isOldEnforced=true`

Seems it is not a good name, I'll change it to `isPrevEnforced`, what do you think？

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 142 / 149 (95.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|65|67|97.01%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|30|30|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 142 / 149 (95.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|65|67|97.01%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|30|30|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 142 / 149 (95.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|65|67|97.01%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|30|30|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/EnforceAndCostTask.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/EnforceAndCostTask.java]: updated

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 136 / 143 (95.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|65|67|97.01%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|24|24|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 136 / 143 (95.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|17|20|85.00%|[49, 52, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|65|67|97.01%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|24|24|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.2 branch-2.3 branch-2.4

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 118 / 125 (94.40%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|14|17|82.35%|[40, 43, 57]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|51|53|96.23%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|24|24|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
BE Build Success

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 117 / 124 (94.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/OutputPropertyGroup.java|14|17|82.35%|[40, 43, 57]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|14|16|87.50%|[171, 172]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|50|52|96.15%|[44, 54]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/GroupExpression.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|24|24|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**